src/iosrv/init.c

```diff
@@ -202,9 +208,12 @@ dss_tgt_nr_get(int ncores, int nr)
 	D_ASSERT(ncores >= 1);
 	/* Each system XS uses one core, and each main XS with
 	 * dss_tgt_offload_xs_nr offload XS. Calculate the nr_default
-	 * as the number of main XS based on number of cores.
+	 * as the number of main XS based on number of cores available
+	 * on this NUMA node.  Reduce the number of cores available
+	 * by the starting core offset.
 	 */
-	nr_default = (ncores - dss_sys_xs_nr) / DSS_XS_NR_PER_TGT;
+	nr_default = (((ncores - dss_sys_xs_nr) - dss_core_offset) /
+		DSS_XS_NR_PER_TGT);
```

Comment:
Another style nit - second line of this statement should be lined up with the first opening parenthesis. You can use spaces if necessary.